File: app/src/main/java/com/partyconsole/companion/model/Item.kt
package com.partyconsole.companion.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement

/**
 * Mirrors party-console's `Item` type (dashboard/features/party/item.tsx).
 * Field names match the server's JSON exactly - these are also the same
 * field names Adventure Land's own game client uses for an item, which
 * party-console's coordinator passes through mostly as-is.
 */
@Serializable
data class Item(
    val l: JsonElement? = null, // string | boolean - locked, or a lock reason
    val gift: Boolean? = null,
    val expires: JsonElement? = null, // string | number
    val name: String,
    val level: Int? = null,
    val q: Int? = null, // quantity, for stackable items
    val p: String? = null, // special/"shiny" variant, e.g. "glitched", "lucky"
    @SerialName("stat_type") val statType: String? = null,
    val price: Long? = null,
    val rid: String? = null,
    val b: Boolean? = null,
    val m: JsonElement? = null, // boolean | string | number
)

@Serializable
data class InventoryEntry(
    val slot: Int,
    val item: Item,
)

@Serializable
data class EquippedEntry(
    val item: Item,
)

/**
 * Mirrors `Condition` (dashboard/features/party/condition.tsx) - one live
 * status effect (e.g. a buff/debuff) on a character.
 */
@Serializable
data class Condition(
    val id: String,
    val name: String,
    val explanation: String? = null,
    val remainingMs: Long? = null,
    val stacks: JsonElement? = null, // number | string
    val source: JsonElement? = null, // string | number
)

File: web/src/models/item.ts
/** Mirrors the Android app's model/Item.kt, itself a mirror of
 *  party-console's `Item` type (dashboard/features/party/item.tsx) -
 *  field names match the server's JSON exactly. `unknown` is used in
 *  place of Kotlin's JsonElement for fields with more than one possible
 *  wire shape (string|boolean|number); callers narrow as needed. */
export interface Item {
  l?: unknown // string | boolean - locked, or a lock reason
  gift?: boolean
  expires?: unknown // string | number
  name: string
  level?: number
  q?: number // quantity, for stackable items
  p?: string // special/"shiny" variant, e.g. "glitched", "lucky"
  stat_type?: string
  price?: number
  rid?: string
  b?: boolean
  m?: unknown // boolean | string | number
}

export interface InventoryEntry {
  slot: number
  item: Item
}

export interface EquippedEntry {
  item: Item
}

/** Mirrors `Condition` (dashboard/features/party/condition.tsx) - one live
 *  status effect (buff/debuff) on a character. */
export interface Condition {
  id: string
  name: string
  explanation?: string
  remainingMs?: number
  stacks?: unknown
  source?: unknown
}
